docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,82 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, slack, or any other method with the owners of this repository before making a change. 
+
+## Conventions and rules
+
+While developing we use the following conventions and rules to develop components.
+
+### Packages
+- *asc-assets* contains icons and fonts. This package can be used standalone in other projects that
+  are not using Amsterdam Styled Components.
+- *asc-core* contains re-exported styled components and theme configuration, that can be found in 
+the `asc-core/src/` folder.
+- *asc-ui* is the main package where we develop our components.
+
+### File structure and content
+A typical structure of a component would look like this
+```
++-- asc-ui
+|   +-- src
+|   |   +-- components
+|   |   |   +-- Footer (1)
+|   |   |   |   +-- Footer.tsx (2)
+|   |   |   |   +-- FooterStyle.ts (3)
+|   |   |   |   +-- FooterStyle.test.tsx (4)
+|   |   |   |   +-- FooterStyle.stories.tsx (5)
+|   |   |   |   +-- index.ts (6)
+|   |   +-- index.ts (7)
+```
+1. This it the directory containing the component's name
+2. This is the React component. This uses the *Style.ts component
+3. This file contains styled-component related stuff, and **no react** code.    
+4. This contains the tests (obviously)
+5. Stories for storybook. Feel free to also add a real life example of how you'd use it in your
+   project!
+6. This will contain a default export, pointed to the .tsx file (2). If needed, you can also 
+   re-export the *Style.ts file.
+7. Eventually, we need to make our component available to use in other project, so we import
+   the corresponding component's default index.ts and make it available in the export.
+
+### Internals 
+`asc-ui/src/internals` folder where the styles are allowed. The components from this folder are 
+just for internal use in the stories.
+
+**Pro tip!** Use generators to quickly add the needed files to develop components.
+To make life a little easier, we added [hygen](https://www.hygen.io/). Simply run 
+`hygen component new`. You will be prompted with questions the generator needs to build the files 
```

Comment:
Added a script in the `package.json` for this in https://github.com/Amsterdam/amsterdam-styled-components/pull/90